rocky/rocky/scheduler.py

```diff
@@ -196,30 +198,43 @@ def list_tasks(
         **kwargs,
     ) -> PaginatedTasksResponse:
         kwargs = {k: v for k, v in kwargs.items() if v is not None}  # filter Nones from kwargs
-        res = self._client.get("/tasks", params=kwargs)
+        endpoint = "/tasks"
+        logger.debug("Making request to %s with args %s", endpoint, kwargs)
+        # Endpoint expects a str.
+        res = self._client.post(endpoint, params=kwargs, data=json.dumps(kwargs.get("filters", None)))  # type: ignore
```

Comment:
`filters` seems to be already passed in `params`, but it's also sent within the request body too?

Reply:
`data` should be a mapping, not a `str` or `bytes` input

For binary request data, such as JSON, use `content` instead of `data` (which is intended for form-encoded data) and set the correct content type